### PR TITLE
Allow migrations to skip postgres or mongo or vault

### PR DESCRIPTION
**Pull request by ryanwohara**
SERVER-1288

Some customers may only have Postgres externalized
and need help migrating. DCEs would have to do the
process manually without this script.

### CODE REVIEW

File: migrate/2.19-export.sh [hunk 1]
```diff
@@ -28,27 +28,29 @@ source 2.19-preflight.sh
 source 2.19-postgres.sh
 source 2.19-mongo.sh
 source 2.19-vault.sh
+source 2.19-key.sh
+source 2.19-compress.sh
```
Comment (robinsturmeit) [inline migrate/2.19-export.sh:32]: That file doesn't seem to exist !?

File: migrate/3.0-restore.sh [hunk 2]
```diff
@@ -79,24 +95,40 @@ function circleci_database_import() {
...
+    if [ ! "$SKIP_VAULT" = "true" ];
+    then
+        VAULT_BU="${BACKUP_DIR}/circleci-vault"
+        import_vault
+    fi
 
-    reinject_bottoken
+    if [ ! "$SKIP_MONGO $SKIP_POSTGRES $SKIP_VAULT" = "true true true" ];
+    then
+        reinject_bottoken
```
Comment (robinsturmeit) [inline migrate/3.0-restore.sh:127]: Wouldn't it be better to move the `reinject_bottoken` step into the block that imports the Mongo data?

File: migrate/migrate.sh [hunk 3]
```diff
@@ -50,25 +89,47 @@ init_options() {
...
+    read -p 'SSH Username: ' USERNAME
+fi
+
 HOST="${USERNAME}@${HOSTNAME}"
 
-echo "Now we need the namespace that has CircleCI Server 3.0 installed."
-read -p 'Namespace: ' NAMESPACE
+if [ -z $NAMESPACE ];
+then
+    echo "Now we need the namespace that has CircleCI Server 3.0 installed."
```
Comment (robinsturmeit) [inline migrate/migrate.sh:130]: This needs to be Server 3.1, or maybe just Server 3 ?

**Review by robinsturmeit — CHANGES_REQUESTED** [review]

File: migrate/migrate.sh (hunk 3 above)
Comment (ryanwohara) [reply migrate/migrate.sh:130]: Ah thanks, been trying to replace it with 3.x but the script was initially written for 3.0.

File: migrate/3.0-restore.sh (hunk 2 above)
Comment (ryanwohara) [reply migrate/3.0-restore.sh:127]: I'm honestly not that familiar with inject bottoken but I'll take your word on it

File: migrate/2.19-export.sh (hunk 1 above)
Comment (ryanwohara) [reply migrate/2.19-export.sh:32]: Yikes. They were local.

**Review by robinsturmeit — APPROVED** [review]
This looks good to me now

**Review by lloydChris — APPROVED** [review]

**Comment by soulchips** [comment]
I found errors when trying to apply the skip database flags.
whether i uses --skip-databases or --skip-postgres, the script still looks for postgres data to migrate.

I also found that i have to pass the host via --hostname. If i dont pass this flag, the script does not request i provide the hostname and will try to connect without it which naturally fails but because of a timeout.

**Pull request merged**